bokehjs/src/coffee/models/annotations/color_bar.coffee

```diff
@@ -0,0 +1,401 @@
+_ = require "underscore"
+
+Annotation = require "./annotation"
+LinearMapper = require "../mappers/linear_mapper"
+LogMapper = require "../mappers/log_mapper"
+Range1d = require "../ranges/range1d"
+BasicTicker = require "../tickers/basic_ticker"
+BasicTickFormatter = require "../formatters/basic_tick_formatter"
+GuideRenderer = require "../renderers/guide_renderer"
+Renderer = require "../renderers/renderer"
+p = require "../../core/properties"
+SidePanel = require "../../core/layout/side_panel"
+{get_text_height} = require "../../core/util/text"
+
+class ColorBarView extends Annotation.View
+  initialize: (options) ->
+    super(options)
+    @_set_image_data()
+
+  _get_panel_offset: () ->
+    # ColorBars draw from the top down, so set the y_panel_offset to _top
+    x = @model.panel._left._value
+    y = @model.panel._top._value
+    return {x: x, y: -y}
+
+  _get_size: () ->
+    geom = @compute_legend_bbox()
+    side = @model.panel.side
+    if side == 'above' or side == 'below'
+      return geom.height
+    if side == 'left' or side == 'right'
+      return geom.width
```

Comment:
You could use `switch` statement instead:
```
size = switch side
  when 'above', 'below'
    geom.height
  when 'left', 'right'
    geom.width
return size
```
(you can return immediately if you prefer so; given coffescript having "expression statements", I prefer to assign first and then have one return path).